src/core/providers/ogr/qgsogrconnpool.cpp

```diff
@@ -18,6 +18,43 @@
 
 #include "qgslogger.h"
 
+
+void QgsOgrConnPoolGroup::connectionCreate( const QString &connInfo, QgsOgrConn *&c )
+{
+  c = new QgsOgrConn;
+
+  const QVariantMap parts = QgsOgrProviderMetadata().decodeUri( connInfo );
+  const QString fullPath = parts.value( QStringLiteral( "vsiPrefix" ) ).toString()
+                           + parts.value( QStringLiteral( "path" ) ).toString()
+                           + parts.value( QStringLiteral( "vsiSuffix" ) ).toString();
+  const QStringList openOptions = parts.value( QStringLiteral( "openOptions" ) ).toStringList();
+  char **papszOpenOptions = nullptr;
+  for ( const QString &option : openOptions )
+  {
+    papszOpenOptions = CSLAddString( papszOpenOptions,
+                                     option.toUtf8().constData() );
+  }
+  c->ds = QgsOgrProviderUtils::GDALOpenWrapper( fullPath.toUtf8().constData(), false, papszOpenOptions, nullptr );
+  CSLDestroy( papszOpenOptions );
+  c->path = connInfo;
+  c->valid = true;
+}
+
+void QgsOgrConnPoolGroup::connectionDestroy( QgsOgrConn *c )
```

Comment:
same (I won't repeat for the other occurrences)
```suggestion
void QgsOgrConnPoolGroup::connectionDestroy( QgsOgrConn *connection )
```